tempodb/encoding/vparquet/wal_block.go

```diff
@@ -446,6 +446,7 @@ func (b *walBlock) Iterator() (common.Iterator, error) {
 }
 
 func (b *walBlock) Clear() error {
+	b.file.Close()
```

Comment:
This is the thing that I think may fix the original issue. `b.writer` is not closed too, but as far as I see, that it is based on the `file`, so there is no reason to double close